File: sdks/csharp/sdk/src/PogodocApi/Templates/Requests/GenerateTemplatePreviewsRequest.cs
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

[Serializable]
public record GenerateTemplatePreviewsRequest
{
    /// <summary>
    /// Type of template to be rendered
    /// </summary>
    [JsonPropertyName("type")]
    public required GenerateTemplatePreviewsRequestType Type { get; set; }

    /// <summary>
    /// Sample data for the template
    /// </summary>
    [JsonPropertyName("data")]
    public Dictionary<string, object?> Data { get; set; } = new Dictionary<string, object?>();

    /// <summary>
    /// Format options for the rendered document
    /// </summary>
    [JsonPropertyName("formatOpts")]
    public GenerateTemplatePreviewsRequestFormatOpts? FormatOpts { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/sdk/src/Templates/Requests/GenerateTemplatePreviewsRequest.php
<?php

namespace Pogodoc\Templates\Requests;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Templates\Types\GenerateTemplatePreviewsRequestType;
use Pogodoc\Core\Json\JsonProperty;
use Pogodoc\Core\Types\ArrayType;
use Pogodoc\Templates\Types\GenerateTemplatePreviewsRequestFormatOpts;

class GenerateTemplatePreviewsRequest extends JsonSerializableType
{
    /**
     * @var value-of<GenerateTemplatePreviewsRequestType> $type Type of template to be rendered
     */
    #[JsonProperty('type')]
    public string $type;

    /**
     * @var array<string, mixed> $data Sample data for the template
     */
    #[JsonProperty('data'), ArrayType(['string' => 'mixed'])]
    public array $data;

    /**
     * @var ?GenerateTemplatePreviewsRequestFormatOpts $formatOpts Format options for the rendered document
     */
    #[JsonProperty('formatOpts')]
    public ?GenerateTemplatePreviewsRequestFormatOpts $formatOpts;

    /**
     * @param array{
     *   type: value-of<GenerateTemplatePreviewsRequestType>,
     *   data: array<string, mixed>,
     *   formatOpts?: ?GenerateTemplatePreviewsRequestFormatOpts,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->type = $values['type'];
        $this->data = $values['data'];
        $this->formatOpts = $values['formatOpts'] ?? null;
    }
}
